modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/value/ValueProviderMediator.java

```diff
@@ -156,7 +159,7 @@ public ValueProviderMediator(T containerModel, Supplier<MuleContext> muleContext
     return valueSet.stream()
         .map(option -> cloneAndEnrichValue(option, parameters))
         .map(ValueBuilder::build)
-        .collect(toSet());
+        .collect(toCollection(LinkedHashSet::new));
```

Comment:
I don't think this is correct. Yes, once the LinkedHashSet is created, the order will be preserved, but if the ValueProvider returns a plain HashSet, results will be in a different order every time. 
Shouldn't we sort the results from the valueSet too? Just in case the VP does not return an ordered set? 
Maybe this is like this by design and we want to keep the order of the results only if the VP wants them in order. Otherwise, we do not care.

Reply:
Exactly, that's the case. As an initial support with the current API we would like to preserve the order but that only happens if the resolver/provider has returned a sorted set. If not, it is exactly as it is right now, order is not warrantied therefore each call to the resolver/provider would return different orders because it doesn't matter.